bundles/org.openhab.binding.icalendar/src/main/java/org/openhab/binding/icalendar/internal/ICalendarHandlerFactory.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.icalendar.internal;
+
+import static org.openhab.binding.icalendar.internal.ICalendarBindingConstants.*;
+
+import java.util.Collections;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.smarthome.core.events.EventPublisher;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerFactory;
+import org.eclipse.smarthome.io.net.http.HttpClientFactory;
+import org.openhab.binding.icalendar.internal.handler.ICalendarHandler;
+import org.osgi.framework.BundleException;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ICalendarHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Michael Wodniok - Initial contribution
+ * @author Andrew Fiddian-Green - EventPublisher code
+ *
+ */
+@NonNullByDefault
+@Component(configurationPid = "binding.icalendar", service = ThingHandlerFactory.class)
+public class ICalendarHandlerFactory extends BaseThingHandlerFactory {
+
+    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Collections.singleton(THING_TYPE_CALENDAR);
+
+    private final Logger logger = LoggerFactory.getLogger(ICalendarHandlerFactory.class);
+    private @Nullable @Reference HttpClientFactory httpClientFactory;
```

Comment:
Why not pass this in through the ICalendarHandlerFactory constructor so you can make this field final?

Reply:
OK. The `EventPublisher` is currently injected using a setter. Should it also get injected via constructor?

Reply:
Yep, inject is via constructor as well.

Reply:
OK